### PR TITLE
Disallow negative counts conversion factor in Aperture Photometry

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to disallow negative counts conversion factor in Aperture Photometry plugin.

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)? [🐱](https://jira.stsci.edu/browse/JDAT-4702)

### CODE REVIEW

File: jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue [hunk 1]
```diff
@@ -116,6 +116,11 @@
           >
           </v-text-field>
         </v-row>
+        <v-row v-if="counts_factor < 0">
+          <span class="v-messages v-messages__message text--secondary" style="color: red !important">
+            Counts conversion factor cannot be negative.
+          </span>
+        </v-row>
```
Comment (kecnry) [inline jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue:123]: can we use validation rules in the `v-text-field` instead?

```
:rules="[() => counts_factor>=0 || 'Counts conversion factor cannot be negative.']"
```

Comment (kecnry) [reply jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue:123]: and while we're at it, might as well disable the action button by adding another term to the already existing `:disabled` in `plugin-action-button`

Comment (pllim) [reply jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.vue]: Sure. Done and added you as co-author. Thanks for the suggestions!

File: jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py [hunk 2]
```diff
@@ -637,7 +637,9 @@ def calculate_photometry(self, dataset=None, aperture=None, background=None,
                     include_pixarea_fac = True
             if img_unit != u.count:
                 try:
-                    ctfac = float(counts_factor if counts_factor is not None else self.counts_factor)  # noqa
+                    ctfac = float(counts_factor if counts_factor is not None else self.counts_factor)  # noqa: E501
+                    if ctfac < 0:
+                        raise ValueError('Counts conversion factor cannot be negative.')
```
Comment (kecnry) [inline jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py:642]: is this by design that the user will actually see the other error (I see the test does test for that, so I'm assuming its intended, just want to double check)?

**Review by kecnry — APPROVED** [review]
Just one clarification question, otherwise looks good to me, thanks!

File: jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py (hunk 2 above)
Comment (pllim) [reply jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py:642]: The design is so `except` catches it. The message does not propagate to user based on current logic. I thought about modifying that logic but that seems out of scope.

**Comment by pllim** [comment]
Thanks for the reviews!

**Pull request merged**